service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -39,30 +45,88 @@
     private final DigitalTwinsImpl _protocolLayer;
     private static final ObjectMapper objectMapper = new ObjectMapper();
 
-    DigitalTwinAsyncClient(String connectionString) {
-    ServiceConnectionString serviceConnectionString = ServiceConnectionStringParser.parseConnectionString(connectionString);
-    SasTokenProvider sasTokenProvider = serviceConnectionString.createSasTokenProvider();
-    String httpsEndpoint = serviceConnectionString.getHttpsEndpoint();
-    final SimpleModule stringModule = new SimpleModule("String Serializer");
-    stringModule.addSerializer(new DigitalTwinStringSerializer(String.class, objectMapper));
-
-    JacksonAdapter adapter = new JacksonAdapter();
-    adapter.serializer().registerModule(stringModule);
-    RestClient simpleRestClient = new RestClient.Builder()
-        .withBaseUrl(httpsEndpoint)
-        .withCredentials(new ServiceClientCredentialsProvider(sasTokenProvider))
-        .withResponseBuilderFactory(new ServiceResponseBuilder.Factory())
-        .withSerializerAdapter(adapter)
-        .build();
-
-    IotHubGatewayServiceAPIsImpl protocolLayerClient = new IotHubGatewayServiceAPIsImpl(simpleRestClient);
-    _protocolLayer = new DigitalTwinsImpl(simpleRestClient.retrofit(), protocolLayerClient);
+    /**
+     * Creates an implementation instance of {@link DigitalTwins} that is used to invoke the Digital Twin features
+     *
+     * @param connectionString The IoTHub connection string
+     * @return The instantiated DigitalTwinAsyncClient.
+     */
+    public DigitalTwinAsyncClient(String connectionString) {
+        ServiceConnectionString serviceConnectionString = ServiceConnectionStringParser.parseConnectionString(connectionString);
+        SasTokenProvider sasTokenProvider = serviceConnectionString.createSasTokenProvider();
+        String httpsEndpoint = serviceConnectionString.getHttpsEndpoint();
+        final SimpleModule stringModule = new SimpleModule("String Serializer");
+        stringModule.addSerializer(new DigitalTwinStringSerializer(String.class, objectMapper));
+
+        JacksonAdapter adapter = new JacksonAdapter();
+        adapter.serializer().registerModule(stringModule);
+        RestClient simpleRestClient = new RestClient.Builder()
+            .withBaseUrl(httpsEndpoint)
+            .withCredentials(new ServiceClientCredentialsProvider(sasTokenProvider))
+            .withResponseBuilderFactory(new ServiceResponseBuilder.Factory())
+            .withSerializerAdapter(adapter)
+            .build();
+
+        IotHubGatewayServiceAPIsImpl protocolLayerClient = new IotHubGatewayServiceAPIsImpl(simpleRestClient);
+        _protocolLayer = new DigitalTwinsImpl(simpleRestClient.retrofit(), protocolLayerClient);
+    }
+
+    /**
+     * Creates an implementation instance of {@link DigitalTwins} that is used to invoke the Digital Twin features
+     *
+     * @param hostName The hostname of your IoT Hub instance (For instance, "your-iot-hub.azure-devices.net")
```

Comment:
does it have to have the `https://` prefix in there? I know that ADT client doesn't like it when the prefix is missing.

Reply:
In ADT land, we ask for the "endpoint", not the hostname, which is why we require the "https://" prefix in ADT. For these clients, I opted to ask for the hostname instead, for consistency with the other service clients in this SDK. Since the HTTP pipeline we use in the digital twin clients still needs an endpoint rather than a hostname, I've made it so the SDK pre-pends the hostname with "https://" in order to turn the given hostname into an endpoint